### PR TITLE
Ad user story 29

### PR DESCRIPTION
**Pull request by adumortier**
- added a test for the `order` `show` page
- added extra information to the `order` `show` page
- added a conditional for the `cart` show page to limit access to `checkout` link to registered users

### CODE REVIEW

File: app/views/orders/show.html.erb [hunk 1]
```diff
@@ -32,17 +38,21 @@
     <tr>
     <section id = "item-<%=item_order.item_id%>">
         <td><p><%=link_to item_order.item.name, "/items/#{item_order.item_id}"%></p></td>
+        <td><p><%= item_order.item.description %> </p></td>
         <td><p><%= link_to item_order.item.merchant.name, "/merchants/#{item_order.item.merchant.id}"%></p></td>
         <td><p><%= number_to_currency(item_order.price)%></p></td>
         <td><p><%= item_order.quantity%></p></td>
-        <td><p><%= number_to_currency(item_order.subtotal)%></p></td>
+        <td><p><%= number_to_currency(item_order.subtotal) %></p></td>
       </section>
```
Comment (TravisBorgsmiller) [inline app/views/orders/show.html.erb:46]: Good job going through the item_orders table

File: spec/features/orders/new_spec.rb [hunk 2]
```diff
@@ -49,6 +63,18 @@
...
+                                  state: "New York",
+                                  zip_code: 10001,
+                                  email: "erniesroomie@example.com",
+                                  password: "paperclips800",
+                                  role: 0)
+
+      allow_any_instance_of(ApplicationController).to receive(:current_user).and_return(@default_user)
+
       visit "/cart"
       click_on "Checkout"
```
Comment (TravisBorgsmiller) [inline spec/features/orders/new_spec.rb:80]: Thanks for fixing this

**Review by TravisBorgsmiller — APPROVED** [review]

File: spec/features/orders/show_spec.rb [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+      expect(page).to have_content(order1.city)
+      expect(page).to have_content(order1.state)
+      expect(page).to have_content(order1.zip)
+
+      within("section#item-#{@item1.id}") do 
+        expect(page).to have_content(@item1.name)
+        expect(page).to have_content(@item1.description)
+        expect(page).to have_content(2)
+        expect(page).to have_content("#{ActiveSupport::NumberHelper.number_to_currency(@item1.price)}")
+        expect(page).to have_content("#{ActiveSupport::NumberHelper.number_to_currency(@item1.price * 2)}")
```
Comment (cgaddis36) [inline spec/features/orders/show_spec.rb:112]: If you wouldn't mind going over this piece that would be very helpful.... Not  sure what this ActiveSupport::NumberHelper is exactly

**Review by cgaddis36 — APPROVED** [review]
Overall everything looks good. Ready for merge

**Pull request merged**